platforms/nuttx/src/px4/microchip/mpfs/px4io_serial/px4io_serial.cpp

```diff
@@ -81,6 +81,9 @@ int ArchPX4IOSerial::init_uart()
 	modifyreg32(MPFS_SYSREG_BASE + MPFS_SYSREG_SUBBLK_CLOCK_CR_OFFSET,
 		    0, SYSREG_SUBBLK_CLOCK_CR_FIC3);
 
+	/* Release IP reset */
```

Comment:
Just add the #ifdef CONFIG_ here as well to check if reset controller is configured for the sw?

Reply:
Hmm, the dependency is not to the SW component, it is to the FPGA being used. The reset MUST be released if an FPGA with the reset controller is used. An option would be to add one extra Kconfig flag like "MPFS_FPGA_RESET_CONTROLLER=y/n" ?

Reply:
One option of course would be to use the rcc driver here as well, is that opaque enough ?

Reply:
Done